### ISSUE TITLE
Align text

### ISSUE THREAD

**Issue by grydstedt** [draft-js-plugins/draft-js-plugins#941]
Can the Alignment plugin be used to left/right/center align text? If I add the AlignBlockCenterButton etc buttons to the static toolbar and try to align text I get `this.props.setAlignment is not a function`

**Comment by nickvoronin** [comment]
@okonet Hi, could you please provide any information about this one?

**Comment by williamcp** [comment]
@nikgraf help us please :)

**Comment by weasteam** [comment]
any solutions? @grydstedt 

I got the same problem:

```
Uncaught TypeError: _this.props.setAlignment is not a function
    at BlockAlignmentButton._this.activate (bundle.js:19195)
    at Object.ReactErrorUtils.invokeGuardedCallback (bundle.js:19888)
    at executeDispatch (bundle.js:19672)
    at Object.executeDispatchesInOrder (bundle.js:19695)
    at executeDispatchesAndRelease (bundle.js:15722)
    at executeDispatchesAndReleaseTopLevel (bundle.js:15733)
    at Array.forEach (<anonymous>)
    at forEachAccumulated (bundle.js:28111)
    at Object.processEventQueue (bundle.js:15933)
    at runEventQueueInBatch (bundle.js:51468)
```

**Comment by grydstedt** [comment]
No solutions yet

**Comment by neo182** [comment]
I am having the same problem for two days in row. Anyone to rescue?

**Comment by josemontiel** [comment]
Same happening here.

**Comment by parris** [comment]
I have a solution I'm using in my codebase for this. I'd make a pull request, but I'm a bit busy at the moment.

Basically I make a new type of toolbar button that updates the data object. It's pretty general purpose and allows you to update anything in the data attribute. It will not work like a "toggle" though, for this use case that's not needed anyways.

```
const createUpdateBlockPropsButton = ({ blockProps, children }) => (
  class BlockPropsButton extends React.Component {

    setProperties = (event) => {
      event.preventDefault();
      const editorState = this.props.getEditorState();
      const content = editorState.getCurrentContent();
      const data = content
        .getBlockForKey(editorState.getSelection().getStartKey())
        .getData();

      const newState = EditorState.push(
        editorState,
        Modifier.mergeBlockData(content, editorState.getSelection(), blockProps),
        'change-block-data',
      );

      this.props.setEditorState(newState);
    }

    preventBubblingUp = (event) => { event.preventDefault(); }

    buttonIsActive = () => {
      // if the button is rendered before the editor
      if (!this.props.getEditorState) {
        return false;
      }

      const editorState = this.props.getEditorState();
      const data = editorState
        .getCurrentContent()
        .getBlockForKey(editorState.getSelection().getStartKey())
        .getData()
        .toJS();

      let isActive = true;

      Object.keys(blockProps).forEach((key) => {
        if (blockProps[key] !== data[key]) {
          isActive = false;
        }
      });

      return isActive;
    }

    render() {
      const { theme } = this.props;
      const className = this.buttonIsActive() ? classnames(theme.button, theme.active) : theme.button;
      return (
        <div
          className={theme.buttonWrapper}
          onMouseDown={this.preventBubblingUp}
        >
          <button
            className={className}
            onClick={this.toggleStyle}
            type="button"
            children={children}
          />
        </div>
      );
    }
  }
);
```

I use it like so:

```
const toolbarPlugin = createToolbarPlugin({
  structure: [
    createUpdateBlockPropsButton({
      blockProps: { alignment: 'left' },
      children: (<FormatAlignLeftIcon />),
    }),
    createUpdateBlockPropsButton({
      blockProps: { alignment: 'center' },
      children: (<FormatAlignCenterIcon />),
    }),
    createUpdateBlockPropsButton({
      blockProps: { alignment: 'right' },
      children: (<FormatAlignRightIcon />),
    }),
 ]
});
```

The only other pieces needed are the renderings of your text, for that I created a plugin that just has a decorator:

```
const alignmentPlugin = {
  decorators: [{
    strategy: (contentBlock, callback) => {
      const text = contentBlock.get('text');
      callback(0, text.length);
    },
    component: ({ children }) => {
      const data = children[0].props.block.getData().toJS();
      return (
        <div
          style={{ textAlign: getAttr(data, 'alignment', 'center') }}
        >
          {children}
        </div>
      );
    },
  }],
};
```

I just have 1 problem remaining. When you create a new line (a new block) data won't get copied over. Does anyone know how to copy data from the previous block when pressing enter?

**Comment by parris** [comment]
Ok figured it out - the following works now:
```
function insertNewLine(editorState) {
  const newEditorState = editorState;
  const contentState = newEditorState.getCurrentContent();
  const selectionState = newEditorState.getSelection();
  const currentBlock = contentState.getBlockForKey(selectionState.getFocusKey());

  const splitBlock = Modifier.splitBlock(contentState, selectionState);
  const newBlock = splitBlock.getBlockAfter(currentBlock.getKey());
  const oldBlockData = currentBlock.getData();
  const newContent = splitBlock.setIn(['blockMap', newBlock.getKey(), 'data'], oldBlockData)

  return EditorState.push(newEditorState, newContent, 'insert-fragment');
}

const alignmentPlugin = {
  handleReturn: (event, editorState, { setEditorState }) => {
    setEditorState(insertNewLine(editorState));
    return 'handled';
  },
  decorators: [{
    strategy: (contentBlock, callback) => {
      const text = contentBlock.get('text');
      callback(0, text.length);
    },
    component: ({ children }) => {
      const data = children[0].props.block.getData().toJS();
      return (
        <div
          style={{ textAlign: getAttr(data, 'alignment', 'center') }}
        >
          {children}
        </div>
      );
    },
  }],
};
```

If you have other plugins with 'onReturn' methods this may not work for you and you'll need some exceptions potentially.

**Comment by juliankrispel** [comment]
I created a new issue to implement this feature, high time it was implemented please check here to follow progress #1104
